pkg/reconciler/taskrun/metrics.go

```diff
@@ -314,11 +318,21 @@ func (r *Recorder) RecordPodLatency(pod *corev1.Pod, tr *v1beta1.TaskRun) error
 		return err
 	}
 
-	metrics.Record(ctx, podLatency.M(float64(latency)))
+	metrics.Record(ctx, podLatency.M(float64(latency/time.Second)))
 
 	return nil
 }
 
+func (r *Recorder) MarkRunningTime(pod *corev1.Pod, runningTime metav1.Time) error {
```

Comment:
So with this implementation, the runAt time is effectively only stored in this map, in memory. If the controller restarts (e.g., a panic, a node failure, an upgrade), that value is lost, and metrics will either report no data, or incorrect data.

To mitigate this, we can rely on the global resync that happens when a controller restarts, to re-populate the value, but that might be tricky too, and tricky to test since it relies on causing unusual state to exercise the code path.